screens/CampsiteInfoScreen.js

```diff
@@ -1,19 +1,50 @@
-import { FlatList, StyleSheet, Text, View } from 'react-native';
+import { FlatList, StyleSheet, Text, View, Button, Modal } from 'react-native';
 import { useSelector, useDispatch } from 'react-redux';
 import RenderCampsite from '../features/campsites/RenderCampsite';
-import { toggleFavorite } from '../features/favorites/favoritesSlice';
+import { toggleFavorite } from '../features/favorites/favoritesSlice'
+import { useState } from 'react';
+import { Rating, Input } from 'react-native-elements';
+import { postComment } from '../features/comments/commentsSlice';
 
 const CampsiteInfoScreen = ({ route }) => {
-    const { campsite } = route.params;
     const comments = useSelector((state) => state.comments);
     const favorites = useSelector((state) => state.favorites);
+    const [showModal, setShowModal] = useState(false);
+    const [rating, setRating] = useState(5);
+    const [author, setAuthor] = useState('');
+    const [text, setText] = useState('');
+
     const dispatch = useDispatch();
+    const { campsite } = route.params;
+
+    const handleSubmit = () => {
+        const newComment = {
+            author,
+            rating,
+            text,
+            campsiteId: campsite.id
+        };
+        console.log(newComment);
```

Comment:
since you've implemented the functionality of this, go ahead and remove this console.log()